includes/server/class-llms-rest-memberships-controller.php

```diff
@@ -0,0 +1,590 @@
+		);
+
+		return $query_params;
+	}
+
+	/**
+	 * Get action/filters to be removed before preparing the item for response.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Membership $membership Membership object.
+	 * @return array Array of action/filters to be removed for response.
+	 */
+	protected function get_filters_to_be_removed_for_response( $membership ) {
+		if ( ! llms_blocks_is_post_migrated( $membership->get( 'id' ) ) ) {
+			return array();
+		}
+
+		return array(
+			// hook => [callback, priority].
+			'lifterlms_single_membership_after_summary' => array(
+				// Membership Information.
+				array(
+					'callback' => 'lifterlms_template_pricing_table',
+					'priority' => 10,
+				),
+			),
+		);
+	}
+
+	/**
+	 * Get the Membership's schema, conforming to JSON Schema.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = parent::get_item_schema();
+
+		$schema['properties']['auto_enroll']         = array(
+			'description' => __(
+				'List of courses to automatically enroll students into when they\'re enrolled into the membership.',
+				'lifterlms' ),
+			'type'        => 'array',
+			'default'     => [],
+			'items'       => array(
+				'type' => 'integer',
+			),
+		);
+		$schema['properties']['catalog_visibility']  = array(
+			'description' => __( 'Visibility of the membership in catalogs and search results.', 'lifterlms' ),
+			'type'        => 'string',
+			'enum'        => array_keys( llms_get_product_visibility_options() ),
+			'default'     => 'catalog_search',
+			'context'     => array( 'view', 'edit' ),
+		);
+		$schema['properties']['categories']          = array(
+			'description' => __( 'List of membership categories.', 'lifterlms' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'integer',
+			),
+			'context'     => array( 'view', 'edit' ),
+		);
+		$schema['properties']['instructors']         = array(
+			'description' => __(
+				'List of post instructors. Defaults to current user when creating a new post.',
+				'lifterlms' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'integer',
+			),
+			'context'     => array( 'view', 'edit' ),
+		);
+		$schema['properties']['restriction_action']  = array(
+			'description' => __(
+				'Determines the action to take when content restricted by the membership is accessed by a non-member. ' .
+				'- `none`: Remain on page and display the message `restriction_message`. ' .
+				'- `membership`: Redirect to the membership\'s permalink. ' .
+				'- `page`: Redirect to the permalink of the page identified by `restriction_page_id`. ' .
+				'- `custom`: Redirect to the URL identified by `restriction_url`.',
+				'lifterlms' ),
+			'type'        => 'string',
+			'default'     => 'none',
+			'enum'        => array( 'none', 'membership', 'page', 'custom' ),
+			'context'     => array( 'view', 'edit' ),
+		);
+		$schema['properties']['restriction_message'] = array(
```

Comment:
I treat the default differently here (so that it's present in the json returned at wp-json/llms/v1/) but we can keep them the way you do, although I think we should replace `id="1234"` with `id="{{membership_id}}`.
See my comment below in `update_additional_object_fields()`.

Reply:
I like using the placeholder in the schema's default value. That way we can use it when creating if `restriction_message` is not given in the request.

Reply:
Yes, that's how I use it, just as default for the whole `restriction_message` object rather than for the `raw` subprop.

Reply:
I think I like your way better with the default so that it shows up in `wp-json/llms/v1/`.